src/test/java/com/example/produktapi/service/ExceptionHandlerAdviceTest.java

```diff
@@ -0,0 +1,67 @@
+package com.example.produktapi.service;
+
+import com.example.produktapi.exception.BadRequestException;
+import com.example.produktapi.exception.EntityNotFoundException;
+import com.example.produktapi.model.Product;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+public class ExceptionHandlerAdviceTest {
+
+    private ExceptionHandlerAdvice exceptionHandlerAdvice;
+
+    @BeforeEach
+    public void setup() {
+        exceptionHandlerAdvice = new ExceptionHandlerAdvice();
+    }
+
+    @Test
+    @DisplayName("unitTestofHandleEntityNotFoundException")
+    void handleEntityNotFoundException() {
+        //Arrange
+        EntityNotFoundException entityNotFoundException = new EntityNotFoundException(0);
+        //Act
+        ResponseEntity<String> response = exceptionHandlerAdvice.handleException(entityNotFoundException);
+        //Assert
+        assertAll(
+                () ->assertEquals(HttpStatus.NOT_FOUND, response.getStatusCode()),
+                () ->assertEquals("Produkt med id 0 hittades inte", response.getBody())
+        );
+    }
+
+    @Test
+    @DisplayName("unitTestofHandleBadRequestException")
```

Comment:
Läsbarheten av detta `DisplayName` skulle kunna förbättras. Syftet med DisplayName är att det ska vara lättare att förstå testets syfte. Således föreslår jag att ändra det till:
```suggestion
    @DisplayName("handleException (BadRequestException)")
```

I denna föreslagna ändring så framkommer det tydligt vilken funktion som testas samt vad den testas mot.

Reply:
Samma som ovan.